server/app/app.py

```diff
@@ -36,6 +36,10 @@ def dataset_index(dataset=None):
 
     try:
         cache_manager = current_app.matrix_data_cache_manager
+        loader = MatrixDataLoader(location)
```

Comment:
why not just have the MatrixDataLoader enforce the data format restrictions, and treat "disabled" types the same as "unsupported" types?   I'm concerned that this feature (enable/disable formats) belongs lower, in a single place, so any other future callers benefit from the config.

Reply:
put another way, why not just have MatrixDataLoader return an "invalid" object whenever the location points at something unsupported/not-understood/disabled?   Or even throw an exception if the location is invalid/disabled/unsupported.

Reply:
Good suggestion.  The code is simpler now.